### PR TITLE
[XPU] Add manual_seed and synchronize method

### PR DESCRIPTION
**Pull request by Stonepia**
This PR set the following device-specific settings for xpu(Intel GPU) specific:
1. Set the manual seed for xpu
2. Set the synchronization method for xpu

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -3687,9 +3693,9 @@ def run(runner, args, original_dir=None):
             log.warning("torch.cuda.is_available() == False, using CPU")
             args.devices = ["cpu"]
 
-    if args.devices != ["cpu"] and torch.cuda.is_available():
+    if args.devices != ["cpu"] and (torch.cuda.is_available() or torch.xpu.is_available()):
```
Comment (EikanWang) [inline benchmarks/dynamo/common.py]:
```suggestion
    if args.devices != ["cpu"] and (HAS_CUDA or HAS_XPU):
        device_type = "cuda" if HAS_CUDA else "xpu" if HAS_XPU else ""
```

File: benchmarks/dynamo/common.py [hunk 2]
```diff
@@ -325,23 +325,6 @@ def nothing(f):
...
-        seed = 1337
-        import torch.cuda
-
-        if not torch.cuda._is_in_bad_fork():
-            torch.cuda.manual_seed_all(seed)
-        return default_generator.manual_seed(seed)
-
-    torch.manual_seed = deterministic_torch_manual_seed
-
-
```
Comment (EikanWang) [inline benchmarks/dynamo/common.py:344]: Is the code block useless?

Comment (Stonepia) [reply benchmarks/dynamo/common.py:344]: Yes, That code seems due to historical reason. The torch.manual_seed already has this logic:

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (Stonepia) [reply benchmarks/dynamo/common.py]: Thanks! The this is changed in https://github.com/pytorch/pytorch/pull/124709/commits/a0b82c81301a0d819a9589f3c0961b91d4c5eeca

File: benchmarks/dynamo/common.py (hunk 2 above)
Comment (Stonepia) [reply benchmarks/dynamo/common.py]: Since this PR should not do big changes. This change is reverted.

**Review by EikanWang — APPROVED** [review]

**Review by desertfire — APPROVED** [review]
